src/uct/ib/rc/base/rc_ep.h

```diff
@@ -424,24 +424,47 @@ uct_rc_ep_fm(uct_rc_iface_t *iface, uct_ib_fence_info_t* fi, int flag)
     /* a call to iface_fence increases beat, so if endpoint beat is not in
      * sync with iface beat it means the endpoint did not post any WQE with
      * fence flag yet */
-    fence          = (fi->fence_beat != iface->tx.fi.fence_beat) ? flag : 0;
-    fi->fence_beat = iface->tx.fi.fence_beat;
+    fence        = (fi->fence_sn != iface->tx.fi.fence_sn) ? flag : 0;
+    fi->fence_sn = iface->tx.fi.fence_sn;
     return fence;
 }
 
 static UCS_F_ALWAYS_INLINE ucs_status_t
-uct_rc_ep_fence(uct_ep_h tl_ep, uct_ib_fence_info_t* fi, int fence)
+uct_rc_ep_fence_add(uct_ep_h tl_ep, uct_ib_fence_info_t* fi, int fence)
 {
     uct_rc_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_rc_iface_t);
 
     /* in case if fence is requested and enabled by configuration
      * we need to schedule fence for next RDMA operation */
     if (fence && (iface->config.fence_mode != UCT_RC_FENCE_MODE_NONE)) {
-        fi->fence_beat = iface->tx.fi.fence_beat - 1;
+        fi->fence_sn = iface->tx.fi.fence_sn - 1;
     }
 
-    UCT_TL_EP_STAT_FENCE(ucs_derived_of(tl_ep, uct_base_ep_t));
     return UCS_OK;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+uct_rc_ep_fence_set(uct_ep_h tl_ep, uct_ib_fence_info_t* fi, int fence)
```

Comment:
it is not same: ```fence_add``` - doesn't clear fence state, in case if ```fence==0``` - it just doesn't change fence state, like logic ```OR``` operation